File: Dairyncia-Spring/src/main/java/com/dairyncia/entities/Farmer.java
package com.dairyncia.entities;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "farmers")
@Getter @Setter @NoArgsConstructor @AllArgsConstructor @Builder @ToString(callSuper=true,exclude={"milkCollections", "user", "address"})
public class Farmer extends BaseEntity{


    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false, unique = true)
    private User user;

    @Column(name = "address_id")
    private Long addressId;

    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "address_id", insertable = false, updatable = false)
    private Address address;

    @OneToMany(mappedBy = "farmer", cascade = CascadeType.ALL)
    @Builder.Default
    private List<MilkCollection> milkCollections = new ArrayList<>();

    @Column(nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
    }
}

File: Dairyncia-dotnet/Models/Farmer.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
namespace Dairyncia.Models
{
    public class Farmer : BaseEntity
    {
        [Required(ErrorMessage = "UserId is required to link farmer with identity user.")]
        public string UserId { get; set; }
        [ForeignKey(nameof(UserId))]
        public ApplicationUser? User { get; set; }

       
        public int? AddressId { get; set; }
        
        [ForeignKey(nameof(AddressId))]
        public Address? Address { get; set; }

        [ForeignKey(nameof(UserId))]
        public string ManagerId { get; set; }

        public ICollection<MilkCollection>? MilkCollections { get; set; }
    }
}
